lib/reply.js

```diff
@@ -33,46 +33,31 @@ Reply.prototype.send = function (payload) {
     this.res.log.warn(new Error('Reply already sent'))
     return
   }
-
   this.sent = true
 
   if (payload instanceof Error || this._isError === true) {
     handleError(this, payload, onSendHook)
     return
   }
 
-  if (payload === undefined) {
-    onSendHook(this, payload)
-    return
-  }
-
-  const contentType = this.res.getHeader('Content-Type')
-  const contentTypeNotSet = contentType === undefined
-
-  if (payload !== null) {
-    if (Buffer.isBuffer(payload) || typeof payload.pipe === 'function') {
+  if (payload !== undefined) {
+    const contentType = this.res.getHeader('Content-Type')
+    const contentTypeNotSet = contentType === undefined
+    if (isPayloadStream(payload)) {
```

Comment:
can you restore this check inline? It increases our inline count for no particular reason.

Reply:
Done